### PR TITLE
fix: add review doesn't behave as expected

### PR DESCRIPTION
**Pull request by elysee15**
I figured out that the review note creation doesn't behave as I expected then I made this fix to make it work with redux correctly

### CODE REVIEW

**Review by bbengfort — COMMENTED** [review]
@elysee15 I've tested the note functionality as follows:

1. Went to a page with no note and checked that it said "No Reviewer Notes"
2. Created a note, left the page and went back to make sure there was no "No Reviewer Notes" text and the note was displayed.
3. Updated the note, navigated to the dashboard and back again.
4. Added another note, updated the first note.

Everything works as expected; however I'm concerned that these changes seem to indicate a workflow that was in error that I didn't try above. Could you indicate how I could create the problem that you're fixing?

Would it be possible to add a story for review notes tests to make sure that things work as expected? 

Finally would you mind setting the text area for the review note [rows=3 instead of 7](https://github.com/trisacrypto/directory/blob/main/web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNoteForm.js#L44) and sending me a screenshot? The review notes text area seems a little large right now and dominates the space, I'd be interested to see what it looks like in a bit more compact of a form.

**Comment by elysee15** [comment]
( old way ) Here, I make a request to have the list of all the notes since Patrick did not return me the created note that's why I asked him to do it. You can see a loader due to `loading` state change
https://user-images.githubusercontent.com/54010836/142156986-b03c1bfb-79d1-47c1-be69-f79e410d7ed3.mov

( the fix ) Here, instead of making a query to get the notes list I push the note returned after creation in the reducer which avoids a reloading of the note list. We no longer have a `loading` state change 
https://user-images.githubusercontent.com/54010836/142157153-24f4cbdc-4b68-4917-86de-390c247235b7.mov

**Comment by elysee15** [comment]
> @elysee15 I've tested the note functionality as follows:
> 
>     1. Went to a page with no note and checked that it said "No Reviewer Notes"
> 
>     2. Created a note, left the page and went back to make sure there was no "No Reviewer Notes" text and the note was displayed.
> 
>     3. Updated the note, navigated to the dashboard and back again.
> 
>     4. Added another note, updated the first note.
> 
> 
> Everything works as expected; however I'm concerned that these changes seem to indicate a workflow that was in error that I didn't try above. Could you indicate how I could create the problem that you're fixing?
> 
> Would it be possible to add a story for review notes tests to make sure that things work as expected?
> 
> Finally would you mind setting the text area for the review note [rows=3 instead of 7](https://github.com/trisacrypto/directory/blob/main/web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNoteForm.js#L44) and sending me a screenshot? The review notes text area seems a little large right now and dominates the space, I'd be interested to see what it looks like in a bit more compact of a form.


Screen Shot 2021-11-17 at 07 57 03

**Comment by bbengfort** [comment]
> ( old way ) Here, I make a request to have the list of all the notes since Patrick did not return me the created note that's why I asked him to do it. You can see a loader due to `loading` state change https://user-images.githubusercontent.com/54010836/142156986-b03c1bfb-79d1-47c1-be69-f79e410d7ed3.mov
> 
> ( the fix ) Here, instead of making a query to get the notes list I push the note returned after creation in the reducer which avoids a reloading of the note list. We no longer have a `loading` state change https://user-images.githubusercontent.com/54010836/142157153-24f4cbdc-4b68-4917-86de-390c247235b7.mov

Excellent, thank you for the explanation! I tested locally and I don't see the loading message.

**Comment by bbengfort** [comment]
@elysee15 your change looks very good and I like it a lot, but I'm not seeing it in this PR:

Screen Shot 2021-11-18 at 17 18 15

I think we might be having rebasing issues that's causing trouble. Same with #176

**Comment by bbengfort** [comment]
Also would  you please change the text in the [deletion confirm alert](https://github.com/trisacrypto/directory/blob/main/web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNote.js#L17) to:

"Are you sure you want to delete this reviewer's note?"

File: web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNote.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function ReviewNote({ note, vaspId }) {
 
     const handleDeleteClick = () => {
         if (note && vaspId) {
-            if (window.confirm('Do you really want to delete it ?')) {
+            if (window.confirm("Are you sure you want to delete this reviewer's note ?")) {
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNote.js:17]:
```suggestion
            if (window.confirm("Are you sure you want to delete this reviewer's note?")) {
```

File: web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNoteForm.js [hunk 2]
```diff
@@ -41,7 +45,7 @@ function ReviewNoteForm({ vaspId }) {
                 className="form-control form-control-light mb-2"
                 placeholder="Write a review note"
                 id="review-note"
-                rows="7"></textarea>
+                rows="3"></textarea>
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/details/ReviewNotes/ReviewNoteForm.js:48]: Thank you!

**Review by bbengfort — APPROVED** [review]
Cheers!

**Pull request merged**